source/extensions/priv/server/ntds_jet.c

```diff
@@ -0,0 +1,363 @@
+	if (dateResult == 0){
+		strncpy(userAccount->logonDate, "Never", 6);
+	}
+	dateResult = GetTimeFormat(LOCALE_SYSTEM_DEFAULT, 0, &lastLogon2, NULL, userAccount->logonTime, 255);
+	if (dateResult == 0){
+		strncpy(userAccount->logonTime, "Never", 6);
+	}
+	// Grab the last password change date and time
+	readStatus = JetRetrieveColumn(ntdsState->jetSession, ntdsState->jetTable, accountColumns->lastPasswordChange.columnid, &lastPass, sizeof(lastPass), &columnSize, 0, NULL);
+	if (readStatus != JET_errSuccess){
+		return readStatus;
+	}
+	//Convert the FILETIME to a SYSTEMTIME so we can get a human readable date
+	FileTimeToSystemTime(&lastPass, &lastPass2);
+	dateResult = GetDateFormat(LOCALE_SYSTEM_DEFAULT, DATE_LONGDATE, &lastPass2, NULL, userAccount->passChangeDate, 255);
+	// Getting Human Readable will fail if account has never logged in, much like the expiry date
+	if (dateResult == 0){
+		strncpy(userAccount->passChangeDate, "Never", 6);
+	}
+	dateResult = GetTimeFormat(LOCALE_SYSTEM_DEFAULT, 0, &lastPass2, NULL, userAccount->passChangeTime, 255);
+	if (dateResult == 0){
+		strncpy(userAccount->passChangeTime, "Never", 6);
+	}
+	// Grab the Account Description here
+	readStatus = JetRetrieveColumn(ntdsState->jetSession, ntdsState->jetTable, accountColumns->accountDescription.columnid, &userAccount->accountDescription, sizeof(userAccount->accountDescription), &columnSize, 0, NULL);
+	if (readStatus == JET_wrnColumnNull){
+		memset(userAccount->accountDescription, 0, sizeof(userAccount->accountDescription));
+	}
+	else if (readStatus != JET_errSuccess){
+		return readStatus;
+	}
+	// Grab the UserAccountControl flags here
+	readStatus = JetRetrieveColumn(ntdsState->jetSession, ntdsState->jetTable, accountColumns->accountControl.columnid, &accountControl, sizeof(accountControl), &columnSize, 0, NULL);
+	if (readStatus != JET_errSuccess){
+		return readStatus;
+	}
+	if (accountControl & NTDS_ACCOUNT_DISABLED){
+		userAccount->accountDisabled = TRUE;
+	}
+	if (accountControl & NTDS_ACCOUNT_LOCKED){
+		userAccount->accountLocked = TRUE;
+	}
+	if (accountControl & NTDS_ACCOUNT_NO_PASS){
+		userAccount->noPassword = TRUE;
+	}
+	if (accountControl & NTDS_ACCOUNT_PASS_EXPIRED){
+		userAccount->passExpired = TRUE;
+	}
+	if (accountControl & NTDS_ACCOUNT_PASS_NO_EXPIRE){
+		userAccount->passNoExpire = TRUE;
+	}
+	// Grab the Logon Count here
+	readStatus = JetRetrieveColumn(ntdsState->jetSession, ntdsState->jetTable, accountColumns->logonCount.columnid, &userAccount->logonCount, sizeof(userAccount->logonCount), &columnSize, 0, NULL);
+	if (readStatus != JET_errSuccess){
+		return readStatus;
+	}
+	// Grab the NT Hash
+	readStatus = JetRetrieveColumn(ntdsState->jetSession, ntdsState->jetTable, accountColumns->ntHash.columnid, encryptedNT, sizeof(encryptedHash), &columnSize, 0, NULL);
+	if (readStatus != JET_errSuccess){
+		if (readStatus == JET_wrnColumnNull){
+			memcpy(&userAccount->ntHash, &BLANK_NT_HASH, 32);

```

Comment:
the '&' is not needed for either argument here. strings and arrays are already pointers

Also, the #defines for BLANK_* really don't need to be in the header file - they only seem to be used once, here.